std/node/string_decoder_test.ts

```diff
@@ -0,0 +1,117 @@
+import { assertEquals } from "../testing/asserts.ts";
+import Buffer from "./buffer.ts";
+import { StringDecoder } from "./string_decoder.ts";
+
```

Comment:
Could you please bring in these tests too:

https://github.com/nodejs/node/blob/master/test/parallel/test-string-decoder-end.js
https://github.com/nodejs/node/blob/master/test/parallel/test-string-decoder-fuzz.js

As well as https://github.com/nodejs/node/blob/f902170af6e19ab0f2d5cefdf7111580238cbeb7/test/parallel/test-string-decoder.js#L28-L35

Reply:
Let's just land this for now, since this PR has been sitting around for a while. We can iterate on it from here.